### DISCUSSION TITLE
channel.queryInfo()

### DISCUSSION THREAD

**Discussion by uscffk** [hyperledger/fabric-sdk-node#670] [Q&A] [answered]
Excuse me,I hava a problem:
 await gateway.connect(ccp, connectionOptions);

    // Get the network (channel) our contract is deployed to.
    const network = await gateway.getNetwork('mychannel');

    // Get the channel object from the network.
    const channel = network.getChannel();

    // Get the latest block from the channel.
    const block = await channel.queryInfo();
    const currentBlockHash = block.currentBlockHash.toString('hex');

the error is “TypeError: channel.queryInfo is not a function”

In the new SDK, has the method of obtaining the hash value of the latest block been deprecated?

**Comment by bestbeforetoday** [comment, marked as answer]
The low-level API from the fabric-client package was completely re-worked for the v2.2 release. The low-level implementation underpinning the fabric-network API is now contained in the fabric-common package, and its Channel implementation does not include a queryInfo() method:

https://hyperledger.github.io/fabric-sdk-node/main/Channel.html

As an alternative, it is possible to use the fabric-network API to **evaluate** the `GetChainInfo` transaction function on the `qscc` system chaincode. This returns a serialized [BlockchainInfo](https://hyperledger.github.io/fabric-protos/protos.html#common-BlockchainInfo) protocol buffer message, which contains the same information as the olf queryInfo() method. You can deserialize this message using the Fabric protocol buffer bindings published in the [fabric-protos](https://hyperledger.github.io/fabric-protos/) package.

If you are just moving up from Fabric v1.4 now, you really should look at migrating to Fabric v2.5 LTS and the newer [Fabric Gateway client API](https://hyperledger.github.io/fabric-gateway/). Here the approach would be the same - to evaluate the `qscc` system chaincode's `GetChainInfo` transaction function.

**Reply by uscffk** [reply]
Thank you for your help! Combined with your suggestion, I solved it by the following way.
  ```
  const channelName = 'mychannel';
  const network = await gateway.getNetwork(channelName);
  const contract = network.getContract('qscc');
  const resultByte = await contract.evaluateTransaction('GetChainInfo', channelName);
  const protobuf = require('protobufjs');
  const myObjectProto = protobuf.loadSync('./fabric-protos/common/ledger.proto');
  const MyObject = myObjectProto.lookupType('common.BlockchainInfo'); 
  const myObject = MyObject.decode(resultByte); 
  logger.info(`Latest block: ${JSON.stringify(myObject)}`);
```
